### DISCUSSION TITLE
Wheel emulation?

### DISCUSSION THREAD

**Discussion by AtlasSkyholder** [berarma/oversteer#128] [Ideas]
Just a question if there is a chance that in the future there would be an option for wheel emulation?

I ask this because of console emulation, specifically PCSX2 and RPCS3.

PCSX2 had wheel emulation built in their emulator in version 1.6.0 and their more recent nightly builds SSE4 wxWidgets. You can map your wheel to a Driving Force GT, Driving Force Pro and 2 other wheels which was great for playing Gran Turismo 3, 4, Initial D and some other racing games, sure the FF wasn't great but it was good enough to enjoy the game.

RPCS3 has only added support for a few wheels, Logitec G25 or G27 being one of them. I use GIMX on windows to emulate my Fanatec GT DD (previously I had a T300RS) into a G25 so I can play GT5 and GT6 on RPCS3.

Recently, I have discovered software like BetterJoy, DS4Windows that pick up controllers like Switch Joycons, PS3 6 axis, and other controllers and convert them into Xbox360 or PS4 controllers. Linux has AntiMicroX for that purpose too. So it gave me the idea if Wheel emulation could be a thing?
If Oversteer could do it, I think it would definitely make Linux more appealing to those that like racing games and emulation.

**Comment by berarma** [comment]
This could be quite an undertaking. I'm not even sure what would be needed and it's outside the application goals.

Why don't you use GIMX on Linux?

**Reply by AtlasSkyholder** [reply]
Hello, I wasn't aware that Gimx had a linux version, but thank you very much for pointing it out. And now that the PCSX2 emulator has added support again, for many peripherals that it used to support back with version 1.6.0, I guess this only leaves RPCS3 the only reason I wanted a wheel emulator, but I guess there is no point to this anymore.

But thank you for the reply.

**Reply by AtlasSkyholder** [reply]
Hey @berarma, I just wanted to give you a heads up in case you might not be aware, but there seems to be a group of people who are working on adding the Simucube base to Linux, they are working on a driver it seems.

https://community.granitedevices.com/t/simucube-2-on-linux/10166

That would be a nice wheelbase to add to Oversteer.

**Reply by berarma** [reply]
> Hey @berarma, I just wanted to give you a heads up in case you might not be aware, but there seems to be a group of people who are working on adding the Simucube base to Linux, they are working on a driver it seems.
> 
> https://community.granitedevices.com/t/simucube-2-on-linux/10166
> 
> That would be a nice wheelbase to add to Oversteer.

As far as I know this wheel is already working on Linux. It's supported through the HID PID driver included in Linux. This driver might have some issues since it doesn't seem to be used much but it just needs some testing.

Anyway, users prefer to use Raw mode with this wheel in Proton because that way all features in the wheel are available to Windows applications. I think you only need to pass some parameters to Proton to make it work. There are post in the Proton repository about this.

Adding it to Oversteer is easy. See other PR to add other wheels. Test the wheel then create a PR. If you don't know how to do all this just sent me the USB vendor/product ids and I'll add it and you test it later.
